### PR TITLE
Test: Update validation of Pg Deal Bid 

### PR DESCRIPTION
**Pull request by marki1an**
Test for https://github.com/prebid/prebid-server-java/pull/1927

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/pg/PgBidResponseSpec.groovy [hunk 1]
```diff
@@ -173,4 +173,39 @@ class PgBidResponseSpec extends BasePgSpec {
...
+
+    def "PBS should invalidate bidder response when line items sizes empty"() {
+        given: "Bid request"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp[0].banner.format = [Format.defaultFormat]
+        }
+
+        and: "Planner Mock line items with a empty sizes"
+        def plansResponse = PlansResponse.getDefaultPlansResponse(bidRequest.site.publisher.id).tap {
+            lineItems[0].sizes = []
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/pg/PgBidResponseSpec.groovy:185]: The second case was already tested, when we add a line item empty then bid sizes `w`&`h` are empty

File: src/test/groovy/org/prebid/server/functional/tests/pg/PgBidResponseSpec.groovy [hunk 2]
```diff
@@ -173,4 +173,39 @@ class PgBidResponseSpec extends BasePgSpec {
...
+        and: "Set bid response without line items sizes 'h' and 'w' "
+        def bidResponse = BidResponse.getDefaultPgBidResponse(bidRequest, plansResponse)
+        def bid = bidResponse.seatbid[0].bid[0]
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "Sending auction request to PBS"
+        def auctionResponse = pgPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder response is invalid"
+        assert auctionResponse.ext?.debug?.pgmetrics?.responseInvalidated?.size() == plansResponse.lineItems.size()
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/pg/PgBidResponseSpec.groovy:201]: As for me, it's explicit that when we got an exception then bids will dscarded :)

File: src/test/groovy/org/prebid/server/functional/model/response/auction/BidResponse.groovy [hunk 3]
```diff
@@ -34,8 +34,10 @@ class BidResponse implements ResponseModel {
         def bid = bidResponse.seatbid[0].bid[0]
         def lineItem = plansResponse.lineItems[0]
         bid.dealid = lineItem.dealId
-        bid.w = lineItem.sizes[0].w
-        bid.h = lineItem.sizes[0].h
+        if (lineItem.sizes[0] != null) {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/response/auction/BidResponse.groovy:37]: `if (lineItem.sizes)`

**Review by Net-burst — APPROVED** [review]
LGMT with minor remark

File: src/test/groovy/org/prebid/server/functional/model/response/auction/BidResponse.groovy (hunk 3 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/model/response/auction/BidResponse.groovy]: `Done`

**Pull request merged**